src/dataflow-types/src/types.rs

```diff
@@ -229,6 +248,70 @@ pub struct DataflowDescription<P, S = (), T = mz_repr::Timestamp> {
     pub id: uuid::Uuid,
 }
 
+fn any_source_import() -> impl Strategy<Value = (GlobalId, SourceInstanceDesc<CollectionMetadata>)>
+{
+    (
+        any::<GlobalId>(),
+        any::<SourceInstanceDesc<CollectionMetadata>>(),
+    )
+}
+
+fn any_dataflow_index() -> impl Strategy<Value = (GlobalId, (IndexDesc, RelationType))> {
+    (any::<GlobalId>(), any::<IndexDesc>(), any::<RelationType>())
+        .prop_map(|(id, index, typ)| (id, (index, typ)))
+}
```

Comment:
nit: you can write this with [the `prop_compose!` macro](https://docs.rs/proptest/latest/proptest/macro.prop_compose.html) in a (probably) more readable way:

```rust
prop_compose! {
    fn any_dataflow_index()(
        id in any::<GlobalId>(),
        index in any::<IndexDesc>(),
        typ in any::<RelationType>()
    ) -> (GlobalId, (IndexDesc, RelationType)) {
        (id, (index, typ))
    }
}
```